client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -213,7 +213,8 @@ const PatternAssembler = ( {
 	};
 
 	const showNotice = ( action: string, pattern: Pattern ) => {
-		noticeOperations.createNotice( { content: getNoticeContent( action, pattern ) } );
+		noticeOperations.removeNotice( action );
+		noticeOperations.createNotice( { id: action, content: getNoticeContent( action, pattern ) } );
```

Comment:
I think we have to find a way to restart the [timer](https://github.com/Automattic/wp-calypso/blob/1f99ec0c229b9a9d31916c606958e1d1730a7c7b/client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/notices/notices.tsx#L35) so we make sure each notice stays on the screen for 5 seconds.

Reply:
Thanks, I missed this!

However, it turned out that it's not as easy as it sounds. What I am doing here is as follows:

1. Establish a mapping between notice content => 5-second timer to remove the notice
   * we cannot use notice id as it will be the same / we cannot detect whether this notice of this id is newer
1. When we receive a new `noticeList` in the `useEffect()` hook:
   * clear and remove the timers of old notices that are already gone (dismissed or replaced because it has the same id as the new notice)
   * trigger a new timer for new notice; which is a notice with content that is not in the map yet.
  
Note that this logic has one edge case: when a user repeatedly pick the same pattern, the timer will not be reset, as the content is the same :( However, I think this is a very rare case. The code is complicated already and I am reluctant to handle this, as this will completely change the implementation 😭 

Let me know what you think!

Reply:
One thing that adds to the complexity of this solution: The following two lines:

```
noticeOperations.removeNotice( action );
noticeOperations.createNotice( { id: action, content: getNoticeContent( action, pattern 
```

triggers two state changes, but React somehow batches the update, so it only triggers one rerender and consequently one `useEffect()` call will be fired. So, we only receive the final `nodeList`; we don't get the chance the see the `nodeList` version when the replaced notice is removed. (If we did, we would have had a way to clear the timer first, for example)

Reply:
Yep, react will [batch state updates](https://react.dev/learn/queueing-a-series-of-state-updates#react-batches-state-updates) to improve the performance. If you want, you can use [flushSync](https://react.dev/reference/react-dom/flushSync) to flush any updates synchronously but it might have a bad effect on performance.

Reply:
A simple way is to clear the timer here as we're able to get the `noticeList`. Or we can also get rid of `withNotices` as it seems not to meet our requirement.

Reply:
Interesting idea, let me try real quick...

Reply:
Feel free to completely change the implementation to reduce the complexity and... Enjoy the challenge! Happy to have you back @fushar.

Reply:
@arthur791004 @miksansegundo I've kind of reimplemented `withNotices` for our requirements. Now I've removed the `useEffect()` "hack" and I feel this logic is more readable and maintainable. Please check! 😄 

@miksansegundo : I'm happy to be back, too!